modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -153,22 +153,20 @@ private AQueryBuilder q() {
   }
 
   /**
-   * Create an authorization predicate to be used with {@link #isAuthorized(Predicate)},
+   * Create an authorization predicate to be used with {@link #isAuthorized(String, String)},
    * restricting access to the user's organization and the given action.
    *
    * @param action
    *     the action to restrict access to
    */
   private Predicate mkAuthPredicate(final String action) {
     final AQueryBuilder q = q();
-    return $(secSvc.getUser().getRoles())
-        .foldl(q.always().not(),
-               new Fn2<Predicate, Role, Predicate>() {
-                 @Override public Predicate apply(Predicate predicate, Role role) {
-                   return predicate.or(mkSecurityProperty(q, role.getName(), action).eq(true));
-                 }
-               })
-        .and(restrictToUsersOrganization());
+    return secSvc.getUser().getRoles().stream()
+            .filter((role) -> includeUIRoles || !role.getName().startsWith("ROLE_UI_"))
```

Comment:
A less "magic string" way to do this would be to filter on roles which have a role target of ACL

Role.java:
```

  /**
   * The target (intended purpose) of a set of roles
   *  USER - Roles which are assigned to users and/or groups to provide access to capabilities
   *  ACL  - Roles which are used to manage access to resources (Event, Series) in an ACL
   *  ALL  - All roles
   */
  enum Target {
    USER, ACL, ALL;
  }

```

Reply:
Actually the above is a nice idea but not currently workable, because I was mixing up Type and Target.

So roles are defined with a type (internal | external) but Target is only currently used for searching when a provider returns a set of roles.

To get this to work, we'd need to add Target to Role and its implementations and their constructors, which may be useful in future but more work than warranted here.

The other roles that probably never show up in event ACLs are ROLE_API_* and ROLE_CAPTURE_AGENT_*